docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx

```diff
@@ -0,0 +1,108 @@
+
+async getExperiment(): Promise<DynamicConfig>
+getExperiment(): DynamicConfig // New API
+....
+
+```
+
+**Context**
+The initial design of async APIs are ensuring every evaluation can be done properly even when the SDK is not compatible by requesting evaluation from Statsig Server. 
+However, as SDK and evaluation rules become more stable, we rarely see such cases. For the performance of core apis, we make all apis synchronous functions.
+
+**Migration Guide**
+If you are using gate as following way, and you are using **javascript runtime error** will be reported, please audit and update the usage
+If you are using typescript, compile error will be reported.
+Meanwhile, we deprecated functions `*Sync()` for example `checkGateSync()`, please use checkGate() 
+```javascript
+Statsig.checkGate(user, "gate").then(() => {
+  // do somethig
+}).catch() // Compile error if you are using typescript. If you are using javascript, there will be runtime error please audit and 
+```
+If you re using as following ways, still recommend to remove await, but there won't be any error reported.
+``` javascript
+const gate = await Statsig.checkGate(user, "gate")
+```
+#### DataAdapter Cache Key Changes
+**Context**
+To better support using multiple Statsig Instances in one runtime, and also better support over different responses format, for example download_config_spec_v1 vs download_config_spec_v2, we updated the cache keys SDK read and write from to avoid cache key collision. 
+
+**Changes**
+- Change cache key for config_specs from `statsig.cache` to 
+
+ `statsig|v1/download_config_specs|{String(useGzip)}|{SHA256Hashed(secretkey)}`
+
+- Change cache key for get_id_list_sources from `statsig.id_lists` to be:
+
+`statsig|get_id_lists|{String(useGzip)}|{SHA256Hashed(secretkey)}`
+
+- Change cache key for individual id list to be 
+
+`statsig|id_list::${String(idListName)}|{String(useGzip)}|${SHA256Hashed(secretKey)}`
+
+**Migration Guide**
+1. Update your DataAdapter if you are customizing the key and get and read logic 
+2. If you are writing to cache in other places (instead of relying on SDK setting the cache), update the call-site to use the new key
+3. If you are ok with cache miss with the first request, then you don’t need to do anything, SDK will write to cache with new key.
+4. If you don’t want cold cache, please write with the new key before push to production
+
+#### Hash Algorithm Change on getClientInitializeResponse() 
+**Context**
+In order to reduce payload size, we changed default hash algorithm being used for all config names to be djb2 instead of SHA256. 
+
+**Changes**
+Config (gate / experiment / layers) names are now hashed with djb2 algorithm instead of SHA256 by default. If you are specifying what hash algorithm to use, this should not have affect on you. 
+
+**Migration Guide**
+If you are using latest (or relatively new client sdk), client side should automatically supported, no actions needed from you. 
```

Comment:
we should pull these versions and list out all client sdks

Reply:
Make sense. I am thinking about a separate page?